core/ledger/ledger.go

```diff
@@ -190,53 +190,60 @@ func (self *Ledger) AddBlock(block *types.Block, stateMerkleRoot common.Uint256)
 
 func (self *Ledger) ExecuteBlock(b *types.Block) (store.ExecuteResult, error) {
 	if !self.ShardID.IsRootShard() {
-		parentBlock, merkleRoot, err := self.ParentBlockCache.GetBlock(b.Header.ParentHeight)
-		if err != nil {
-			if err == scommon.ErrNotFound {
-				return self.ldgStore.ExecuteBlock(b)
-			} else {
-				log.Errorf("Ledger ExecuteBlock GetBlock sharad height:%d,ParentHeight:%d error:%s", b.Header.Height, b.Header.ParentHeight, err)
-				return store.ExecuteResult{}, err
+		parentHeight := self.ParentLedger.GetCurrentBlockHeight()
+		if parentHeight != b.Header.ParentHeight {
+			parentBlock, merkleRoot, err := self.ParentBlockCache.GetBlock(b.Header.ParentHeight)
+			if err != nil {
+				if err == scommon.ErrNotFound {
+					return self.ldgStore.ExecuteBlock(b)
```

Comment:
Although this from old-code, 
I failed to figure out why executing shard block when failed to get ParentBlock?

Reply:
@xiemylogos